cmd/juju/application/flags.go

```diff
@@ -136,3 +139,70 @@ func (m stringMap) String() string {
 	}
 	return strings.Join(pairs, ";")
 }
+
+// configFlag records k=v attributes from command arguments
+// and also any files specified.
+type configFlag struct {
+	files []string
+	attrs map[string]interface{}
+}
+
+// Set implements gnuflag.Value.Set.
+func (f *configFlag) Set(s string) error {
+	if s == "" {
+		return errors.NotValidf("empty string")
+	}
+	fields := strings.SplitN(s, "=", 2)
+	if len(fields) == 1 {
+		f.files = append(f.files, fields[0])
+		return nil
+	}
+	var value interface{}
+	if fields[1] == "" {
+		value = ""
+	} else {
+		if err := yaml.Unmarshal([]byte(fields[1]), &value); err != nil {
```

Comment:
I don't think we should be parsing the value as YAML for application config. Can we make that bit configurable please, and disable it for the "juju config" command?

Reply:
done